docs/design/core/metrics/Design.md

```diff
@@ -60,107 +60,174 @@ standard metrics collected by the SDK.
+                                     .build());
+```
+
+The methods exposed for setting metric publishers follow the pattern established by `ExecutionInterceptor`s:
+
+```java
+class RequestOverrideConfiguration {
+    // ...
+    class Builder {
+        // ...
+        Builder metricPublishers(List<MetricPublisher> metricsPublishers);
+        Builder addMetricPublisher(MetricPublisher metricsPublisher);
+    }
+}
+```
+
+### Option 2: Configuring MetricPublishers on a client
+
+A publisher can be configured directly on the `ClientOverrideConfiguration`. A publisher specified in this way is used
+with lower priority than **Option 1** above.
 
 ```java
-ClientOverrideConfiguration config = ClientOverrideConfiguration
-    .builder()
-    // If this is not set, SDK uses the default chain with system property
-    .metricConfigurationProvider(new SystemSettingsMetricConfigurationProvider())
-    .build();
-
-// Set the ClientOverrideConfiguration instance on the client builder
-CodePipelineAsyncClient asyncClient =
-    CodePipelineAsyncClient
-        .builder()
-        .overrideConfiguration(config)
-        .build();
+MetricPublisher metricPublisher = CloudWatchMetricPublisher.create();
+DynamoDbClient dynamoDb = DynamoDbClient.builder()
+                                        .overrideConfiguration(c -> c.addMetricPublisher(metricPublisher))
+                                        .build();
 ```
 
-### Metrics Provider Chain
+The methods exposed for setting metric publishers follow the pattern established by `ExecutionInterceptor`s:
+
+```java
+class ClientOverrideConfiguration {
+    // ...
+    class Builder {
+        // ...
+        Builder metricPublishers(List<MetricPublisher> metricsPublishers);
+        Builder addMetricPublisher(MetricPublisher metricsPublisher);
+    }
+}
+```
+
+**Note:** As with the `httpClient` setting, calling `close()` on the `DynamoDbClient` *will not* close the configured
+`metricPublishers`. You must close the `metricPublishers` yourself when you're done using them.
+
+### Option 3: Configuring MetricPublishers using System Properties or Environment Variables
+
+This option allows the customer to enable metric publishing by default, without needing to enable it via **Option 1** 
+or **Option 2** above. This means that a customer can enable metrics without needing to make a change to their runtime 
+code.
+
+This option is enabled using an environment variable or system property. If both are specified, the system property 
+will be used. If metrics are enabled at the client level using **Option 2** above, this option is ignored. Overriding 
+the metric publisher at request time using **Option 1** overrides any publishers that have been enabled globally.
+
+**System Property:** `aws.metricPublishingEnabled=true` 
+
+**Environment Variable:** `AWS_METRIC_PUBLISHING_ENABLED=true`
+
+The value specified must be one of `"true"` or `"false"`. Specifying any other string values will result in 
```

Comment:
Should we make it case insensitive?

Reply:
+1, as we talked about, though, we should use what other booleans do, first and foremost.